charts/trino/values.yaml

```diff
@@ -649,21 +651,43 @@ worker:
   # ```
 
   lifecycle: {}
-  # worker.lifecycle -- To enable [graceful
-  # shutdown](https://trino.io/docs/current/admin/graceful-shutdown.html),
-  # define a lifecycle preStop like bellow, Set the
-  # `terminationGracePeriodSeconds` to a value greater than or equal to the
-  # configured `shutdown.grace-period`. Configure `shutdown.grace-period` in
-  # `additionalConfigProperties` as `shutdown.grace-period=2m` (default is 2
-  # minutes). Also configure `accessControl` because the `default` system
-  # access control does not allow graceful shutdowns.
+  # worker.lifecycle -- Worker container [lifecycle
+  # events](https://kubernetes.io/docs/tasks/configure-pod-container/attach-handler-lifecycle-event/)
   # @raw
   # Example:
   # ```yaml
   #  preStop:
   #    exec:
-  #      command: ["/bin/sh", "-c", "curl -v -X PUT -d '\"SHUTTING_DOWN\"' -H \"Content-type: application/json\" http://localhost:8081/v1/info/state"]
+  #      command: ["/bin/sh", "-c", "sleep 120"]
+  # ```
+  # If provided, it will override the `preStop` lifecycle event configured by `gracefulShutdown`.
+
+  gracefulShutdown:
+    enabled: false
+    gracePeriod: 2m
+    accessControl:
+      user: admin
+      configFile: graceful-shutdown-rules.json
+  # worker.gracefulShutdown -- Configure [graceful
+  # shutdown](https://trino.io/docs/current/admin/graceful-shutdown.html)
+  # @raw
+  # Example:
+  # ```yaml
+  #  gracefulShutdown:
+  #    enabled: true
+  #    gracePeriod: 2m
+  #    accessControl:
+  #      user: admin
+  #      configFile: graceful-shutdown-rules.json
   # ```
+  # Enabling this feature will:
+  #  1) Add a `preStop` lifecycle event to all worker Pods;
+  #  2) Set the `shutdown.grace-period` configuration property to `gracePeriod`;
+  #  3) Configure the workers' `accessControl` since the `default` system access control [does not allow graceful
+  #    shutdowns](https://trino.io/docs/current/admin/graceful-shutdown.html).
+  # The user must set the `terminationGracePeriodSeconds` to a value of at least two times the configured `gracePeriod`.
```

Comment:
I don't like that we can't make this feature complete. It's so easy to miss this note, and we can't really verify that anywhere in the chart - we can't parse intervals to compare them.

We should remove the default, set it based on the grace period (can it parse the same units?), and allow users to override it, same as with the worker `preStop` lifecycle policy.

Reply:
I think we can require the user to provide the two values in seconds (and add the unit suffix when required), and do a sanity check on them with a `fail` if incompatible.